packages/client/lib/client/authx/token-manager.spec.ts

```diff
@@ -0,0 +1,614 @@
+import { strict as assert } from 'node:assert';
+import { Token } from './token';
+import { IDPError, RetryPolicy, TokenManager, TokenManagerConfig, TokenStreamListener } from './token-manager';
+import { FakeClock } from './clock';
+import { IdentityProvider, TokenResponse } from './identity-provider';
+
+describe('TokenManager', () => {
+
+  /**
+   * Helper function to delay execution for a given number of milliseconds.
+   * @param ms
+   */
+  const delay = (ms: number) => {
+    return new Promise(resolve => setTimeout(resolve, ms));
+  }
```

Comment:
async `setTimeout` is now [built into node](https://nodejs.org/docs/latest-v18.x/api/timers.html#timerspromisessettimeoutdelay-value-options), there used to be a utility in redis for this which I think was removed in preference of the built-in functionality which is being used extensively in other tests.

Reply:
Thanks @dhensby! I'll switch to using the built-ins